### PR TITLE
Refactor the `ContainerVerbsTab` class with MVC in mind

### PR DESCRIPTION
**Pull request by madoar**
This PR replaces the `ContainerVerbsTab` class with the two classes `ContainerVerbsPanel` and `ContainerVerbsPanelSkin`, which need to be wrapped inside a `Tab` object.
In addition this PR replaces the `TiledPane` of the `ContainerVerbsTab` with a `GridPane`, that allows for the selection of multiple verbs to install at the same time.

### CODE REVIEW

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/container/control/ContainerVerbsPanel.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+     * A list of all tool scripts
+     */
+    private final ObservableList<ScriptDTO> verbScripts;
+
+    /**
+     * A boolean signifying whether all verb buttons should be locked
+     */
+    private final BooleanProperty lockVerbs;
+
+    public ContainerVerbsPanel(ObjectProperty<ContainerDTO> container, ObjectProperty<VerbsManager> verbsManager,
```
Comment (plata) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/container/control/ContainerVerbsPanel.java]: javadoc

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/container/control/ContainerVerbsPanel.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        return verbScripts;
+    }
+
+    @Override
+    public ContainerVerbsPanelSkin createSkin() {
+        return new ContainerVerbsPanelSkin(this);
+    }
+
+    public ContainerDTO getContainer() {
+        return container.get();
```
Comment (plata) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/container/control/ContainerVerbsPanel.java]: add `this` (also below)

**Comment by madoar** [comment]
This is how the new verbs installation tab looks like now:

grafik

**Comment by plata** [comment]
For me it's a bit strange that every verb has it's own install button additionally to the general install button.

**Comment by madoar** [comment]
I believe it is a question of preferences. In case a user want to install only a single verb it is more intuitive for the user in my opinion to click on the `Install` button for the desired verb, while in case the user wants to install multiple verbs at the same time, it is more intuitive to select them first and then to install them.

**Comment by madoar** [comment]
@qparis what do you think about the visual redesign of the container verbs tab?

**Comment by plata** [comment]
I'm fine with the checkboxes in general. Like I said, I would remove the install buttons because I believe that they are confusing. For example: Do I have to click the checkbox to install a single verb? Probably not. But I have to do so to install multiple verbs. So the same GUI component is used differently depending on the context. That's not a good idea in general.
You can also compare this to software management tools like pamac or muon.

Maybe it would also be a good idea to show a dialog: "Are you sure you want to install these verbs: mspatcha, vcrun2013".

**Comment by madoar** [comment]
Hmm, so do you suggest to only remove the install buttons or do you want to replace them with some other components?

Let's move the suggestion of adding a confirm dialog when installing verbs to a new issue.

**Comment by ImperatorS79** [comment]
One could rename the button "Information" for https://github.com/PhoenicisOrg/scripts/issues/823 ?

**Comment by madoar** [comment]
How/where should the description be shown after the user has pressed on an information button?
In my opinion it is better to show the description as a tooltip when the user hovers his/her mouse on top of a verb name or its checkbox.

**Comment by ImperatorS79** [comment]
Why not. I am just making propositions.

**Comment by madoar** [comment]
The issue with an `Information` button is, that it requires an addition component/dialog which shows the information after clicking on the button, therefore the question how where we would show the information for the verb(s). A tooltip doesn't require an additional component, but has the disadvantage that it is not as intuitive as a button solution.

Again I think we should move the topic of showing the description for the verbs to its own PR.

**Comment by qparis** [comment]
I don't see how it would work if you remove the install button. Can you show me?

**Comment by plata** [comment]
You select the check box and then click "install selected".

**Comment by madoar** [comment]
This is how it looks now:
grafik
grafik

**Comment by qparis** [comment]
I think it is much better

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/container/control/ContainerVerbsPanel.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    public void setVerbs(ApplicationDTO verbs) {
+        this.verbs.set(verbs);
+    }
+
+    public ObservableList<ScriptDTO> getVerbScripts() {
+        return this.verbScripts;
+    }
+
+    public boolean isLockVerbs() {
+        return lockVerbs.get();
```
Comment (plata) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/container/control/ContainerVerbsPanel.java]: `this.lockVerbs` (like below)

Comment (madoar) [reply phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/container/control/ContainerVerbsPanel.java]: I must have overlooked that one...

**Pull request merged**